src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -216,11 +228,20 @@ private static bool ValidateInvocationPropertiesToLog(TContext context, IEnumera
 
         internal AggregatingLogger InitializeLogger(AnalyzeOptionsBase analyzeOptions)
         {
+            _tool = Tool.CreateFromAssemblyData();
+
             var logger = new AggregatingLogger();
 
             if (!analyzeOptions.Quiet)
             {
-                logger.Loggers.Add(new ConsoleLogger(analyzeOptions.Verbose));
+                _consoleLogger = new ConsoleLogger(analyzeOptions.Verbose, _tool.Driver.Name) { CaptureOutput = _captureConsoleOutput };
+                logger.Loggers.Add(_consoleLogger);
+            }
+
+            if (analyzeOptions.ComputeFileHashes)
```

Comment:
>ComputeFileHashes [](start = 31, length = 17)

This option (command line option `--hashes`) is obsolete, replaced by `analyzeOptions.DataToInsert` (command line option `--insert`). I suggest that you move the following code:
```
                        OptionallyEmittedData dataToInsert = analyzeOptions.DataToInsert.ToFlags();

                        // This code is required in order to support the obsoleted ComputeFileHashes argument
                        // on the analyze command-line;
                        if (analyzeOptions.ComputeFileHashes) { dataToInsert |= OptionallyEmittedData.Hashes; }
```
... which currently exists in the `InitializeOutputFile` method, to the top of the `Run` method, store the result in an instance file `_dataToInsert`, and use it wherever you've added code to use `ComputeFileHashes`.

You can also do what we do in `SarifLogger`, namely, introduce a property `AnalysisCommandBase.ComputeFileHashes` that pulls out the appropriate flag.

It's a shame to duplicate that code -- ideally it would be an extension method on `AnalyzeOptionsBase` -- but `AnalyzeOptionsBase` already has a property `ComputeFileHashes`. Maybe you could rename that to `ComputeFileHashesObsolete`.

Reply:
In fact I notice in your PR description that you expect this feature to be driven by `--hashes`. I think you might have forgotten about `--insert`.

---
In reply to: [303215000](https://github.com/microsoft/sarif-sdk/pull/1574#discussion_r303215000) [](ancestors = 303215000)